### ISSUE TITLE
delete and update events may not be correctly sinked to downstream

### ISSUE THREAD

**Issue by zhaoxinyu** [pingcap/tiflow#8420]
### What did you do?

1. run `csv_storage_basic` test 
```
CASE="csv_storage_basic" docker-compose -f ./deployments/ticdc/docker-compose/docker-compose-storage-integration.yml up
```
2. run `sync_diff_inspector` to check difference between upstream and downstream.

### What did you expect to see?

All dml events can be sinked correctly.

### What did you see instead?

Delete and update events are not sinked properly.

### Versions of the cluster

TiCDC version (execute `cdc version`):

```console
v6.7.0-master
```

**Comment by zhaoxinyu** [comment]
This bug also affects MySQL sink, not only storage consumer.

**Comment by zhaoxinyu** [comment]
For MySQL sink, the reproduce steps are as follows:
1. create a changefeed with MySQL sink.
2. execute the following queries in upstream TiDB:
```
use test
CREATE TABLE t0 ( id INT not null, name varchar(128) CHARACTER SET gbk not null, country char(32) CHARACTER SET gbk, city varchar(64), description  text CHARACTER SET gbk, image tinyblob, UNIQUE
KEY (id, name) ) ENGINE = Innodb;

INSERT INTO t0
VALUES (1, '测试', "中国", "上海", "你好,世界"
	, 0xC4E3BAC3CAC0BDE7);

INSERT INTO t0
VALUES (2, '部署', "美国", "纽约", "世界,你好"
	, 0xCAC0BDE7C4E3BAC3);

UPDATE t0
SET name = '开发'
WHERE name = '测试';
```

**We expect to see the following output:**
```
mysql> select * from t0;
+----+--------+---------+--------+---------------+--------------------+
| id | name   | country | city   | description   | image              |
+----+--------+---------+--------+---------------+--------------------+
|  1 | 开发   | 中国    | 上海   | 你好,世界     | 0xC4E3BAC3CAC0BDE7 |
|  2 | 部署   | 美国    | 纽约   | 世界,你好     | 0xCAC0BDE7C4E3BAC3 |
+----+--------+---------+--------+---------------+--------------------+
2 rows in set (0.00 sec)
```

**But we actually got:**
```
mysql> select * from t0;
+----+--------+---------+--------+---------------+--------------------+
| id | name   | country | city   | description   | image              |
+----+--------+---------+--------+---------------+--------------------+
|  1 | 开发   | NULL    | NULL   | NULL          | NULL               |
|  2 | 部署   | 美国    | 纽约   | 世界,你好     | 0xCAC0BDE7C4E3BAC3 |
+----+--------+---------+--------+---------------+--------------------+
2 rows in set (0.00 sec)
```

**Comment by asddongmen** [comment]
After enabling batch DML feature, when converting update statements, TiCDC will transform the statements into the following format:

from:
```
CREATE TABLE t0 ( id INT not null, name varchar(128) CHARACTER SET gbk not null, country char(32) CHARACTER SET gbk, city varchar(64), description  text CHARACTER SET gbk, image tinyblob, UNIQUE
KEY (id, name) ) ENGINE = Innodb;

UPDATE t0
SET name = '开发'
WHERE name = '测试';
```
covert to:
```
"UPDATE `test`.`t0` SET `id`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `name`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `country`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `city`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `description`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `image`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END WHERE ROW(`id`,`name`) IN (ROW(?,?))\"]"] [values="[[1,\"测试\",1,1,\"测试\",\"开发\",1,\"测试\",\"中国\",1,\"测试\",\"上海\",1,\"测试\",\"你好,世界\",1,\"测试\",\"xOO6w8rAvec=\",1,\"测试\"]]"
```

The behavior of this statement differs between TiDB and MySQL. It may cause the update statement of this type to not achieve the expected result when executed on MySQL.
Ref: https://docs.pingcap.com/tidb/stable/sql-statement-update#mysql-compatibility
However, all downstream integration tests of TiCDC currently use TiDB, so relevant issues were not previously detected.

**Comment by asddongmen** [comment]
> After enabling batch DML feature, when converting update statements, TiCDC will transform the statements into the following format:
> 
> from:
> 
> ```
> CREATE TABLE t0 ( id INT not null, name varchar(128) CHARACTER SET gbk not null, country char(32) CHARACTER SET gbk, city varchar(64), description  text CHARACTER SET gbk, image tinyblob, UNIQUE
> KEY (id, name) ) ENGINE = Innodb;
> 
> UPDATE t0
> SET name = '开发'
> WHERE name = '测试';
> ```
> 
> covert to:
> 
> ```
> "UPDATE `test`.`t0` SET `id`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `name`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `country`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `city`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `description`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END, `image`=CASE WHEN ROW(`id`,`name`)=ROW(?,?) THEN ? END WHERE ROW(`id`,`name`) IN (ROW(?,?))\"]"] [values="[[1,\"测试\",1,1,\"测试\",\"开发\",1,\"测试\",\"中国\",1,\"测试\",\"上海\",1,\"测试\",\"你好,世界\",1,\"测试\",\"xOO6w8rAvec=\",1,\"测试\"]]"
> ```
> 
> The behavior of this statement differs between TiDB and MySQL. It may cause the update statement of this type to not achieve the expected result when executed on MySQL. Ref: https://docs.pingcap.com/tidb/stable/sql-statement-update#mysql-compatibility However, all downstream integration tests of TiCDC currently use TiDB, so relevant issues were not previously detected.

This will be fixed by: #8452
